autospec/pkg_integrity.py

```diff
@@ -114,7 +114,22 @@ def __init__(self, pubkey=None, home=None):
 
     def verify(self, _, tarfile, signature):
         """Validate tarfile with signature."""
-        args = self.args + ['--verify', signature, tarfile]
+        # Since autospec can only verify one signature for now, extract the
+        # first signature from the detached signature file.
+        sig_name = signature
+        packets = parse_gpg_packets(signature)
+        if len(packets) > 1:
+            # sig file may be ascii-armored, so dearmor it first...
+            args = self.args + ['--dearmor', '--output', '-', signature]
+            output, err, code = self.exec_cmd(args)
+            if code != 0:
+                return None
+            num_bytes = packets[0]["length"]
```

Comment:
Is it possible for length to not be defined here given how packets are currently added?

Reply:
Yes, good catch. It needs to check if the `length` key is present.